integration_tests/runJest.js

```diff
@@ -6,17 +6,34 @@
  * of patent rights can be found in the PATENTS file in the same directory.
  */
 
-const spawnSync = require('child_process').spawnSync;
+const {fileExists} = require('./utils');
 const path = require('path');
+const spawnSync = require('child_process').spawnSync;
+
 const JEST_PATH = path.resolve(__dirname, '../packages/jest-cli/bin/jest.js');
 
 // return the result of the spawned proccess:
 //  [ 'status', 'signal', 'output', 'pid', 'stdout', 'stderr',
 //    'envPairs', 'options', 'args', 'file' ]
 function runJest(dir, args) {
-  return spawnSync(JEST_PATH, args || [], {
+  const localPackageJson = path.resolve(dir, 'package.json');
+  if (!fileExists(localPackageJson)) {
+    throw new Error(`
+      Make sure you have a local package.json file at
+        "${localPackageJson}".
+      Otherwise Jest will try to traverse the directory tree and find the
+      the global package.json, which will send Jest into infinite loop.
```

Comment:
yeah.. super annoying bug, because you don't see any output and it hangs for forever